src/libraries/Directory.Build.props

```diff
@@ -215,12 +215,14 @@
     
   <!-- Set up Default symbol and optimization for Configuration -->
   <Choose>
-    <When Condition="'$(ConfigurationGroup)'=='Debug'">
+    <When Condition="'$(ConfigurationGroup)' == 'Debug'">
       <PropertyGroup>
         <DebugSymbols Condition="'$(DebugSymbols)' == ''">true</DebugSymbols>
         <Optimize Condition="'$(Optimize)' == ''">false</Optimize>
         <DebugType Condition="'$(DebugType)' == ''">full</DebugType>
-        <DefineConstants>$(DefineConstants),DEBUG,TRACE</DefineConstants>
+        <!-- Test projects should be configuration agnostic. -->
```

Comment:
There are many places in the tests that use `Debug.Assert` to validate internal test assumptions. These `Debug.Asserts` are going to be no-ops with this change because of they are Conditional on DEBUG. There are other methods Conditional on DEBUG that have the same problem. It is not what we want.

Thinking about it some more, removing the `DEBUG` define for debug builds of the tests does not sound like a good idea.

Yes, we do want to make debug builds of the tests to pass on release build of the product, and vice versa. But it is fine to have `DEBUG` defined for the debug builds of the tests and let the tests do extra work when this define is set. It is the same reasoning why we have `DEBUG` defined for the debug builds of the product.

Reply:
Makes sense. I will do that instead. Thanks.